### PR TITLE
Add equals and hashcode method to MemorySize configuration class to fix unnecessary Keycloak Dev Service reloading

### LINKED ISSUE #43701: Keycloak devservice restarts with every update in devmode
### Describe the bug

We have just had Keycloak devservice moved into its own extension, and on `main`, it restarts even if none of its properties have changed. and the demo does not like that.
We probably need do manual property comparison now after Keycloak `DevServicesConfig` got migrated to `ConfigMapping` to have a finer grained control of the process  

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Start for exampe `quarkus-quickstarts/security-openid-connect-quickstart`, note Keycloak has started. Change the code, see Keycloak loading again

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by michalvavrik**
- Closes: https://github.com/quarkusio/quarkus/issues/43701

I have added also hashcode as it seems that configmapping generates hashcode as well https://smallrye.io/smallrye-config/Main/config/mappings/#tostring-equals-hashcode and this way it will be more precise.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySize.java [hunk 1]
```diff
@@ -25,4 +26,19 @@ public long asLongValue() {
...
+            return true;
+        if (object == null || getClass() != object.getClass())
+            return false;
+        MemorySize that = (MemorySize) object;
+        return Objects.equals(value, that.value);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(value);
```
Comment (gastaldi) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySize.java:42]: Store the hashCode as an attribute in the constructor for improved performance

Comment (gsmet) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/MemorySize.java:42]: I don't expect it to be hot path so I would say we can avoid it.

**Review by gsmet — APPROVED** [review]
LGTM! Nice detective work.

**Comment by sberyozkin** [comment]
Proposing a backport to 3.15.x which uses Keycloak 25.x

**Comment by sberyozkin** [comment]
Oh sorry, the backport label is already there

**Pull request merged**